### DISCUSSION TITLE
Form validations settings missing after upgrade

### DISCUSSION THREAD

**Discussion by wei2go** [solspace/craft-freeform#536] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
I have a very old Craft site (3.5) and it's in a process of updates to the latest Craft 3 (3.7), together with all the plugins. Freeform is one of them, being updated from 3.7.1 to 3.13.21. Updates and db migration went smoothly without error. However, I found some forms cannot be saved with the error "Properties for key 'validation' do not contain TYPE". Then I noticed those forms have an empty "Validation" settings tab (normally you will have some settings). Also I found in db, those forms have empty array for the "validation" key under "layoutJson" column. Or some of them doesn't even have a "validation" key. It also only happens to some forms, not all.

It would be great if you can advise what's the best approach to fix it (ideally without rolling back to the previous version and upgrade again).

**Steps to reproduce**

1. Update Freeform from 3.7.1 to 3.13.21
2. Visit some of the forms' builder page
3. Can see the "Validation" tab shows up empty (only text "Please select an element", without any other fields)
4. Try to save the form
5. Fail with error: Properties for key 'validation' do not contain TYPE

**Expected behavior**
Validation settings tab display correctly and forms can be saved successfully.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 3.5.19.1 to Craft Pro 3.7.27.2
- Freeform Version: 3.7.1 to 3.13.21
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Additional context**

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi @wei2go,

We will investigate as to why this would have happened in your case, but meanwhile, the quickest fix without rolling anything back, would be to add default data to the `validation` entry in the `layoutJson` column in all the forms where it is missing.

```json
{
  "composer": {
    "layout": [...],
    "properties": {
      "page0": {...},
      "form": {...},
      
      --->
      "validation": {
        "type": "validation",
        "validationType": "submit",
        "successMessage": "",
        "errorMessage": ""
      },
      --->
      ...
    }
  },
  ...
}
```

Here's a quick example of how the default values would look and where to put them.
